File: javascript/space-age/space-age.js
const earthSeconds = 31557600;
module.exports = class SpaceAge {
    constructor(seconds) {
        this.seconds = seconds;
        this.earthYears = this.seconds / earthSeconds
    }

    static toFixed(input) {
        return +(input.toFixed(2));
    }

    onEarth() {
        return SpaceAge.toFixed(this.earthYears);
    }

    onMercury() {
        return SpaceAge.toFixed(this.earthYears / 0.2408467);
    }

    onVenus() {
        return SpaceAge.toFixed(this.earthYears / 0.61519726);
    }

    onMars() {
        return SpaceAge.toFixed(this.earthYears / 1.8808158);
    }

    onJupiter() {
        return SpaceAge.toFixed(this.earthYears / 11.862615);
    }

    onSaturn() {
        return SpaceAge.toFixed(this.earthYears / 29.447498);
    }

    onUranus() {
        return SpaceAge.toFixed(this.earthYears / 84.016846);
    }

    onNeptune() {
        return SpaceAge.toFixed(this.earthYears / 164.79132);
    }
};

File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {

    private double seconds;
    private final double EARTH_ORBIT = 31557600;
    private final double MERCURY_RATIO = 0.2408467;
    private final double VENUS_RATIO = 0.61519726;
    private final double MARS_RATIO = 1.8808158;
    private final double JUPITER_RATIO =  11.862615;
    private final double SATURN_RATIO = 29.447498;
    private final double URANUS_RATIO = 84.016846;
    private final double NEPTUNE_RATIO = 164.79132;

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double getSeconds() {
        return this.seconds;
    }

    double onEarth() {
        return seconds / EARTH_ORBIT;
    }

    double onMercury() {
        return this.onEarth() / MERCURY_RATIO;
    }

    double onVenus() {
        return this.onEarth() / VENUS_RATIO;
    }

    double onMars() {
        return this.onEarth() / MARS_RATIO;
    }

    double onJupiter() {
        return this.onEarth() / JUPITER_RATIO;
    }

    double onSaturn() {
        return this.onEarth() / SATURN_RATIO;
    }

    double onUranus() {
        return this.onEarth() / URANUS_RATIO;
    }

    double onNeptune() {
        return this.onEarth() / NEPTUNE_RATIO;
    }

}
